agent/engine/execcmd/manager_init_task_linux.go

```diff
@@ -133,6 +141,36 @@ func (m *manager) InitializeContainer(taskId string, container *apicontainer.Con
 	return nil
 }
 
+func (m *manager) getLatestVersionedHostBinDir() (string, error) {
+	versions, err := retrieveAgentVersions(ecsAgentDepsBinDir)
+	if err != nil {
+		return "", err
+	}
+	sort.Sort(sort.Reverse(byAgentVersion(versions)))
+
+	var latest string
+	for _, v := range versions {
+		vStr := v.String()
+		ecsAgentDepsVersionedBinDir := filepath.Join(ecsAgentDepsBinDir, vStr)
+		if !fileExists(filepath.Join(ecsAgentDepsVersionedBinDir, SSMAgentBinName)) {
+			continue // try falling back to the previous version
+		}
+		// TODO: [ecs-exec] This requirement will be removed for SSM agent V2
+		if !fileExists(filepath.Join(ecsAgentDepsVersionedBinDir, SSMAgentWorkerBinName)) {
+			continue // try falling back to the previous version
+		}
+		if !fileExists(filepath.Join(ecsAgentDepsVersionedBinDir, SessionWorkerBinName)) {
+			continue // try falling back to the previous version
+		}
+		latest = filepath.Join(m.hostBinDir, vStr)
+		break
+	}
+	if latest == "" {
```

Comment:
Is there a test for this condition?  I might have missed it.
I'm looking over the pr as a whole and keep thinking about what happens with an existing but empty directory -- the sort will return 0 (I believe -- no test for that case but it makes sense) and latest will be the empty string.  This is a pretty important error path -- if no available ssm agents, then big problems.

Reply:
[These](https://github.com/aws/amazon-ecs-agent/pull/2701/files#diff-ebdff96a4c1a16c1f6b7675e6e244503b94530f6f3f59358c87b42f74b61bb16R78-R88) two test cases cover this